src/load_balancer/load_balancer.rs

```diff
@@ -9,23 +10,31 @@ use hyper::{
 };
 use hyper_util::rt::TokioIo;
 use tokio::net::TcpStream;
+use tokio::sync::RwLock;
 
 use crate::data_store::Worker;
-use crate::utils::{validate_address, DF_TABLE_NAME};
+use crate::utils::{df_to_table, validate_address, DF_TABLE_NAME};
 use crate::BoxBody;
 use crate::error::LoadBalancerError;
 
 // load balancer now holds ctx with registered table with worker hosts information
 pub struct LoadBalancer {
     pub ctx: SessionContext,
     pub current_worker: usize,
+    pub algorithm: Algorithm,
+}
+
+pub enum Algorithm {
```

Comment:
Seeing that we have a `unwrap_or()` inside the new method implementation, it would make sense to use #[derive(Default)] implementation, and set #[default] tag on RoundRobin. This will help remove the unnecessary Option tags. 

This way, when you create a default Algorithm, you can initialize it as `Default::default()` instead.

Reply:
That's a good point!